src/words/environment/Environment.java

```diff
@@ -135,6 +135,24 @@ public WordsObject createObject(String objectName, String className, Position po
 		}
 	}
 	
+	public void removeObject(WordsObject object) {
+		object.clearReferers();
+		
+		for (HashMap<String, WordsObject> scope : objectsByName) {
+			if (scope.get(object.getObjectName()) != null) {
```

Comment:
good call, check out how i'm doing it now. now i just check wether remove() returns null, and break if it doesn't. we should still break at the first scope at which the object is found, right?